format/Message.fbs

```diff
@@ -21,10 +21,69 @@ include "Tensor.fbs";
 
 namespace org.apache.arrow.flatbuf;
 
+/// ------------------------------------------------------
+/// Buffer encoding schemes.
+/// -------------------------------------------------------
+
+/// Encoding for buffers representing integer as offsets from a reference value.
+/// This encoding uses less bits then the logical type indicates.
+/// It saves space when all values in the buffer can be represented with a
+/// small bit width (e.g. if all values in an int64 column are between -128
+/// and 127, then a bit-width of 8 can be be used) offset from the
+/// reference value.
+table FrameOfReferenceIntEncoding {
+  /// The value that all values in the buffer are relative to.
+  reference_value: long = 0;
```

Comment:
Something to think about: this is representing all values as deltas from a flat line.  However, a sloped line is more useful generally.  For example, timestamps represent points on an increasing sloped line.

Reply:
I agree some data looks like a sloped line, I'm not sure if it is more useful generally (maybe domain specific?) Since this is per record batch, this representation allows for step-wise approximation of a sloped lines across record batches.  It also continues to allows for O(1) access to any element in the buffer and easy slicing.   Was there an alternative that you think might work better?

Reply:
Depending on the size of your batch, a sloped representation would result in far smaller arrays, since the delta from a slope is typically much smaller and can fit into less bits.  You can still do O(1) access to any element, just compute ax + b etc.   Assuming the data is actually increasing, of course - otherwise step wise is fine.